experimental/overhead_matching/kimera_spectacular_data_provider.cc

```diff
@@ -0,0 +1,282 @@
+#include "experimental/overhead_matching/kimera_spectacular_data_provider.hh"
+
+#include <gflags/gflags.h>
+// #include <glog/logging.h>
+#include <gtsam/base/Vector.h>
+#include <gtsam/geometry/Cal3DS2.h>
+#include <gtsam/geometry/Pose3.h>
+
+#include <algorithm>  // for max
+#include <fstream>
+#include <map>
+#include <string>
+#include <utility>  // for pair<>
+#include <vector>
+
+#include "kimera-vio/frontend/StereoFrame.h"
+#include "kimera-vio/imu-frontend/ImuFrontend-definitions.h"
+#include "kimera-vio/logging/Logger.h"
+#include "kimera-vio/utils/YamlParser.h"
+
+DEFINE_string(dataset_path, "/Users/Luca/data/MH_01_easy",
+              "Path of dataset (i.e. Euroc, /Users/Luca/data/MH_01_easy).");
+DEFINE_int64(initial_k, 0,
+             "Initial frame to start processing dataset, "
+             "previous frames will not be used.");
+DEFINE_int64(final_k, 100000,
+             "Final frame to finish processing dataset, "
+             "subsequent frames will not be used.");
+
+namespace robot::experimental::overhead_matching {
+
+VIO::Timestamp vio_time_from_robot_time(const time::RobotTimestamp& robot_time) {
+    /// VIO timestamp is int64_t nanoseconds. time::RobotTimestamp is std::nano
+    return robot_time.time_since_epoch().count();
+}
+
+/* -------------------------------------------------------------------------- */
+SpectacularDataProviderInterface::SpectacularDataProviderInterface(const std::string& dataset_path,
+                                                                   const int& initial_k,
+                                                                   const int& final_k,
+                                                                   const VIO::VioParams& vio_params)
+    : DataProviderInterface(),
+      vio_params_(vio_params),
+      dataset_path_(dataset_path),
+      current_k_(std::numeric_limits<VIO::FrameId>::max()),
+      initial_k_(initial_k),
+      final_k_(final_k),
+      imu_measurements_(),
+      spec_log_(dataset_path) {
+    // CHECK(!dataset_path_.empty()) << "Dataset path for SpectacularDataProvider is empty.";
+
+    // Start processing dataset from frame initial_k.
+    // Useful to skip a bunch of images at the beginning (imu calibration).
+    CHECK_GE(initial_k_, 0);
+
+    // Finish processing dataset at frame final_k.
+    // Last frame to process (to avoid processing the entire dataset),
+    // skip last frames.
+
+    if (final_k_ > spec_log_.num_frames()) {
+        LOG(WARNING) << "Provided final frame for KimeraSpectacularDataProvider was above the "
+                        "number of frames. Reducing to match size of dataset";
+        final_k_ = spec_log_.num_frames();
+    }
+
+    CHECK_GT(final_k_, 0);
+
+    CHECK_GT(final_k_, initial_k_)
+        << "Value for final_k (" << final_k_ << ") is smaller than value for"
+        << " initial_k (" << initial_k_ << ").";
+    current_k_ = initial_k_;
```

Comment:
I would say to prefer ROBOT_CHECK() over the GLOG check. This can be written as:
```
ROBOT_CHECK(final_k_ > initial_k_, "Value for final_k is smaller than vallue for initial_k", final_k_, initial_k_);
```